rust/agama-dbus-server/src/network/model.rs

```diff
@@ -81,6 +90,52 @@ impl NetworkState {
         Ok(())
     }
 
+    /// Updates a controller connection with a new one.
+    ///
+    /// It uses the `id` to decide which connection to update.
+    ///
+    /// Additionally, it registers the connection to be removed when the changes are applied.
+    pub fn update_controller_connection(
+        &mut self,
+        mut conn: Connection,
+        settings: HashMap<String, ControllerSettings>,
+    ) -> Result<(), NetworkStateError> {
+        // let Some(old_conn) = self.get_connection_mut(conn.id()) else {
+        //     return Err(NetworkStateError::UnknownConnection(conn.id().to_string()));
+        // };
+        //
+        let mut new_ports = vec![];
+        let controller = conn.clone();
+
+        if let Connection::Bond(ref mut bond) = conn {
+            if let Some(ControllerSettings::Options(opts)) = settings.get("options") {
+                bond.bond.options = BondOptions::try_from(opts.clone()).unwrap()
+            }
+
+            if let Some(ControllerSettings::Ports(ports)) = settings.get("ports") {
+                for port in ports.iter() {
+                    new_ports.push(
```

Comment:
I would reorganize this a bit to avoid a full `if-else` clause as a parameter for `new_ports`. It is hard to read (at least for me, I know, I am getting old).